2.train_model/scripts/nbconverted/train_multi_class_models.py

```diff
@@ -47,104 +48,97 @@
 
 # In[4]:
 
 
-# specify model types and feature types
-model_types = ["final", "shuffled_baseline"]
-feature_types = ["CP", "DP", "CP_and_DP", "CP_zernike_only", "CP_areashape_only"]
-balanced_types = ["balanced", "unbalanced"]
-
-# specify parameters to tune for
 parameters = {"C": np.logspace(-3, 3, 7), "l1_ratio": np.linspace(0, 1, 11)}
 print(f"Parameters being tested during grid search: {parameters}\n")
 
-# train model on each combination of model type, feature type, and balance type
-for balance in balanced_types:
-    # Set sklearn class_weight parameter for balanced or unbalanced models
-    if balance == "balanced":
-        balance_model = "balanced"
-    else:
-        balance_model = None
-        
-    for model_type in model_types:
-        for feature_type in feature_types:
-            
-            if feature_type == "CP_zernike_only":
-                zernike_only = True
-                dataset = "CP"
-            else:
-                zernike_only = False
-                dataset = feature_type
-                
-            if feature_type == "CP_areashape_only":
-                area_shape_only = True
-                dataset = "CP"
-            else:
-                area_shape_only = False
-    
-            print(f"Training {model_type} {balance} model on {feature_type} features with zernike only {zernike_only} or area features only {area_shape_only}...")
-            
-            X, y = get_X_y_data(
-                training_data,
-                dataset,
-                zernike_only,
-                area_shape_only
-            )
+
+# ## Find best parameters, train models
+
+# In[5]:
+
+
+for model_type, feature_type, balance_type, dataset_type in itertools.product(
+    model_types, feature_types, balance_types, dataset_types
+):
+    # print what combination of types we are training model for
+    print(
+        f"Training model for: \nModel Type: {model_type} \nFeature Type: {feature_type} \nBalance Type: {balance_type} \nDataset Type: {dataset_type}"
+    )
     
-            print(f"X has shape {X.shape}, y has shape {y.shape}")
+    # load training data from indexes and features dataframe
+    data_split_path = pathlib.Path(f"../1.split_data/indexes/data_split_indexes__{dataset_type}.tsv")
+    features_dataframe_path = pathlib.Path(f"../0.download_data/data/labeled_data__{dataset_type}.csv.gz")
     
-            # shuffle columns of X (features) dataframe independently to create shuffled baseline
-            if model_type == "shuffled_baseline":
-                for column in X.T:
-                    np.random.shuffle(column)
+    # dataframe with only the labeled data we want (exclude certain phenotypic classes)
+    features_dataframe = get_features_data(features_dataframe_path)
+    data_split_indexes = pd.read_csv(data_split_path, sep="\t", index_col=0)
+
+    # get training data from labeled data
+    training_data = get_dataset(features_dataframe, data_split_indexes, "train")
```

Comment:
quick clarification: is this where the no-ic vs. ic dataset is being specified? I don't see where this variable is specified.

Reply:
oooh, I see it is actually in line 85!